### PR TITLE
Add shared channels on admin console

### PR DESCRIPTION
**Pull request by nronas**
#### Summary
Add's shared channels and remote_cluster_service on license features.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-29027
https://mattermost.atlassian.net/browse/MM-30838

#### Related Pull Requests

- Has server changes https://github.com/mattermost/mattermost-server/pull/16415
- Has redux changes https://github.com/mattermost/mattermost-redux/pull/1305

## Without a license, no entry of `Shared Channels` should be present
Screenshot 2020-11-26 at 12 30 17 PM

## With an E10 license, no entry of `Shared Channels` should be present
Screenshot 2020-11-26 at 12 30 17 PM

## With an E20 license, no entry of `Shared Channels` should be present
Screenshot 2020-11-26 at 12 28 07 PM

⚠️ Missing Translations, since the copy is not finalized ⚠️ 
🛑 This waits on the [shared channels branch](https://github.com/mattermost/mattermost-server/compare/MM-27493_shared_channels) to be merged first 🛑

### CODE REVIEW

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -5338,6 +5338,17 @@ const AdminDefinition = {
...
                     // },
+                    {
+                        type: Constants.SettingsTypes.TYPE_BOOL,
+                        key: 'ExperimentalSettings.EnableSharedChannels',
+                        label: t('admin.experimental.shared_channels.title'),
+                        label_default: 'Shared Channels:',
+                        help_text: t('admin.experimental.shared_channels.desc'),
+                        help_text_default: 'Toggles Shared Channels',
+                        help_text_markdown: false,
+                        isHidden: it.not(it.licensed),
```
Comment (marianunez) [inline components/admin_console/admin_definition.jsx:5349]: Are we OK here to just be licensed for E10 or should it check specifically for shared channels? I'm not familiar with the licensing details of this new feature. 

```suggestion
                        isHidden: it.not(it.licensedForFeature("SharedChannels")),
```

Comment (nronas) [reply components/admin_console/admin_definition.jsx:5349]: You are 💯 % right, it needs to be only enabled under E20

**Comment by nronas** [comment]
@wiersgallak 👋 who's the best person to ask for final copies around admin configuration for `Shared channels`? More specifically we will need copies for the following sections:

```
label: t('admin.experimental.shared_channels.title'),
help_text: t('admin.experimental.shared_channels.desc'),
help_text_default: 'Toggles Shared Channels',
```

Screenshot 2020-11-26 at 12 28 07 PM

**Review by marianunez — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:    I think you can add some placeholder text to make our govet happy while you wait for final copy.

**Comment by wiersgallak** [comment]
@nronas Here is the text: 
label: "Enable Shared Channels:"
help_text / help_text_default: "When true, users who have permission to manage shared channels can configure another Mattermost server to synchronize and share a channel. Disabling shared channels stops synchronizing the channel with the other Mattermost server, however the channel continues to function for local users per normal channel behavior."

**Comment by nronas** [comment]
@lindalumitchell This should be now ready to be tested with the test server.

**Review by furqanmlk — COMMENTED** [review]
I am able to verify `Enable Shared Channels` showing up for E20 only.
However, UI does not match with the screenshot in description
1. Noticed `Enable Shared Channel` instead of `Shared Channel`
2. Default value should be `false` or `true` ?
image

**Comment by nronas** [comment]
@furqanmlk Thanks a lot for looking at this.

Yeah "Enabled Shared Channels" is the right one, the screenshot have been taken before the final copy was provided. The source of truth for the naming should be in @wiersgallak comment here: https://github.com/mattermost/mattermost-webapp/pull/7136#issuecomment-737398832 . Lastly, the default value should be false

**Comment by nronas** [comment]
@furqanmlk is this QA complete? I've listed the expected requirements of this in https://github.com/mattermost/mattermost-webapp/pull/7136#issuecomment-755184271

**Comment by nronas** [comment]
@furqanmlk gentle ping on 

> @furqanmlk is this QA complete? I've listed the expected requirements of this in #7136 (comment)

**Review by furqanmlk — APPROVED** [review]
Thanks @nronas 
All tests passed ✅ 

**With E20 License** 
image

**With E0 License** 
image

**With E10 License** 
image

**Pull request merged**